Migrate to new csproj format

StummeJ opened the pull request:
dotnet core preview 4 sdk switched to a new `csproj` format.  It is next to impossible if not impossible to get the old format to build when you install the preview 4 tools.

It also seems like all of the main line @aspnet projects are on the new format which makes it hard to contribute to both those projects and these.

In reference to issue #139

PinpointTownes commented:
Thanks for taking the time to do that. It looks like the .sln file was not correctly updated, which seems to be the root cause of the build errors.

(please note that I won't merge it before VS 2017 and the new MSBuild-based tooling are officially released)

PinpointTownes commented on samples/Mvc.Client/Mvc.Client.csproj:
```diff
@@ -0,0 +1,123 @@
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.Cookies">
+      <Version>1.0.0</Version>
```
Hum, I thought there was a shorter syntax for that?

```xml
<PackageReference Include="Microsoft.AspNetCore.Authentication.Cookies" Version="1.0.0" />
```

PinpointTownes commented on samples/Mvc.Client/Mvc.Client.csproj:
```diff
@@ -0,0 +1,123 @@
+
+  <PropertyGroup>
+    <TargetFrameworks>net451;netcoreapp1.0</TargetFrameworks>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.23-x64;opensuse.13.2-x64</RuntimeIdentifiers>
```
Hum, the samples are portable apps (at least, currently). Why do we need the RIDs here?

StummeJ commented:
Sounds good. I'll fix the sln.

PinpointTownes commented on src/AspNet.Security.OAuth.Beam/AspNet.Security.OAuth.Beam.csproj:
```diff
@@ -27,28 +18,13 @@
-    <PackageReference Include="Microsoft.AspNetCore.Authentication.OAuth">
-      <Version>1.0.0</Version>
-    </PackageReference>
+    <PackageReference Include="JetBrains.Annotations" Version="10.1.4" />
```
Please note that we want this dependency to be build-time only.

StummeJ commented:
I've migrated everything over to csproj now and fixed the sln.  The migration tool in preview4 had a bug there. We're still getting build errors that look to me more bugs in the toolchain.  I'll look further into this later...

For now: **anyone that is thinking about upgrading to preview4, DONT**

PinpointTownes commented:
> I've migrated everything over to csproj now and fixed the sln.

Thanks! :clap: 

I took a look at the CI logs, and most/all the restore errors seem to be caused by the fact the NuGet `imports` that are needed to make the JetBrains annotations package work with .NET Standard are no longer here.

The good news is that a newer package that supports .NET Standard 1.0 was released in October: https://www.nuget.org/packages/JetBrains.Annotations/10.2.1, so I bet migrating to the latest bits should help get rid of most the errors we're seeing.

StummeJ commented:
Which build log are these?  It looks like even after updating it is still failing.

PinpointTownes commented:
Yeah, but it fails at build time, now: https://ci.appveyor.com/project/aspnet-contrib/aspnet-security-oauth-providers/build/1.0.0-beta3-374

StummeJ commented:
:joy:  Windows is building.  Linux is having issues still... So it  OSX... but this is a good starting point.

StummeJ commented:
Okay so we'll see how this goes.  There was a new project type UUID for preview4.  Once I updated to that and fixed a few things in the props files it started to build most things.  Only thing not building on my machine is the samples.

PinpointTownes commented on build/dependencies.props:
```diff
@@ -0,0 +1,6 @@
+<Project>
+  <PropertyGroup>
+    <NetStandardImplicitPackageVersion>1.6.1</NetStandardImplicitPackageVersion>
```
1.6.1 shipped with .NET Core 1.1.0. Since we target LTS, we probably want 1.6.0 if we decide to explicitly reference `NETStandard.Library`.

PinpointTownes commented on build/dependencies.props:
```diff
@@ -0,0 +1,6 @@
+<Project>
+  <PropertyGroup>
+    <NetStandardImplicitPackageVersion>1.6.1</NetStandardImplicitPackageVersion>
+    <CoreFxVersion>4.3.0</CoreFxVersion>
```
Same remark, we need to target the version that shipped with LTS.

StummeJ replied:
Looks like this will go away as preview4 assumes a version automatically.

PinpointTownes commented on samples/Mvc.Client/Mvc.Client.csproj:
```diff
@@ -0,0 +1,95 @@
+<Project Sdk="Microsoft.NET.Sdk.Web" ToolsVersion="15.0">
+
+  <PropertyGroup>
+    <RuntimeIdentifiers>win7-x64;win7-x86;osx.10.10-x64;osx.10.11-x64;ubuntu.14.04-x64;ubuntu.16.04-x64;centos.7-x64;rhel.7.2-x64;debian.8-x64;fedora.23-x64;opensuse.13.2-x64</RuntimeIdentifiers>
```
Hum, looks like the RIDs reappeared. Are they automagically added by VS?

StummeJ replied:
Yeah VS adds an error when you try to open the sample project without RID.

GuardRex replied:
@StummeJ Try with ...
```
<RuntimeFrameworkVersion>1.1.0</RuntimeFrameworkVersion>
```
I just found out that sample apps I'm working with will restore and compile for both `netcoreapp1.1` and `net451` if the props are set up this way ...
```
<PropertyGroup>
  <TargetFrameworks>net451;netcoreapp1.1</TargetFrameworks>
  <OutputType>Exe</OutputType>
  <RuntimeFrameworkVersion>1.1.0</RuntimeFrameworkVersion>
</PropertyGroup>
```
```
dotnet restore
dotnet publish -r Release -f netcoreapp1.1
dotnet publish -r Release -f net451
```

PinpointTownes commented on samples/Mvc.Client/Mvc.Client.csproj:
```diff
@@ -0,0 +1,95 @@
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="1.0.3" />
```
Make sure we use `1.0.0` everywhere.

PinpointTownes commented on src/AspNet.Security.OAuth.GitHub/project.json:
```diff
@@ -1,54 +0,0 @@
-
-    "netstandard1.3": {
-      "dependencies": {
-        "System.Dynamic.Runtime": "4.0.11"
```
This dependency was necessary to make JSON.NET work with `dynamic`. It looks like it's no longer here in the .csproj. We'll have to make sure it doesn't cause a regression.

PinpointTownes commented:
@StummeJ wooops, looks like your PR was impacted by the version changes in project.json (a new wave was released last week), I'm sorry for that.

FYI, I spent a few hours migrating most the aspnet-contrib repos to .csproj and it looks like everything is now working fine on both AppVeyor and Travis (see https://github.com/aspnet-contrib/AspNet.Security.OAuth.Extensions/pull/63/files for an example).

That would be great if you could rebase your PR on top of the `dev` branch. If you can't, please ping me, I'll cherry-pick your commits and continue locally.

Thanks for your contribution! :clap:

PinpointTownes commented on build.ps1:
```diff
@@ -3,7 +3,7 @@ cd $PSScriptRoot
 $env:REPO_FOLDER = $repoFolder
 
-$koreBuildZip="https://github.com/aspnet/KoreBuild/archive/1.0.0.zip"
+$koreBuildZip="https://github.com/aspnet/KoreBuild/archive/feature/msbuild.zip"
```
FYI, this version is obsolete and will be removed in the future. You need the `dev` version: https://github.com/aspnet-contrib/AspNet.Security.OAuth.Extensions/pull/63/files#diff-f0e02e42e53ec40f5d3b8bed031ad09e

PinpointTownes commented on build.sh:
```diff
@@ -2,7 +2,7 @@
 cd $repoFolder
 
-koreBuildZip="https://github.com/aspnet/KoreBuild/archive/1.0.0.zip"
+koreBuildZip="https://github.com/aspnet/KoreBuild/archive/feature/msbuild.zip"
```
This file must also be updated: https://github.com/aspnet-contrib/AspNet.Security.OAuth.Extensions/pull/63/files#diff-0b83f9dedf40d7356e5ca147a077acb4

PinpointTownes commented on build/common-oauth.props:
```diff
@@ -0,0 +1,22 @@
+    </PropertyGroup>
+
+    <ItemGroup>
+        <EmbeddedResource Include="compiler\resources\**\*" />
```
This doesn't seem to be necessary.

PinpointTownes commented on build/common-oauth.props:
```diff
@@ -0,0 +1,22 @@
+        <PackageReference Include="Microsoft.AspNetCore.Authentication.OAuth" Version="1.0.0" />
+    </ItemGroup>
+
+    <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">
```
Please remove.

PinpointTownes commented on build/common-oauth.props:
```diff
@@ -0,0 +1,22 @@
+
+    <ItemGroup>
+        <ProjectReference Include="../AspNet.Security.OAuth.Extensions/AspNet.Security.OAuth.Extensions.csproj" />
+        <PackageReference Include="JetBrains.Annotations" Version="10.2.1" PrivateAssets="All" />
```
`10.3.0`.

That would be cool if you could use the same pattern as the one I used here: https://github.com/aspnet-contrib/AspNet.Security.OAuth.Extensions/pull/63/files#diff-30dcc9789f8ea3c7632e4f054e60c072

PinpointTownes commented on build/common.props:
```diff
@@ -0,0 +1,20 @@
+  <Import Project="..\version.props" />
+
+  <PropertyGroup>
+    <Product>Microsoft .NET Extensions</Product>
```
`aspnet-contrib`

PinpointTownes commented on version.props:
```diff
@@ -0,0 +1,7 @@
+<Project>
+    <PropertyGroup>
+        <VersionPrefix>1.0.0</VersionPrefix>
+        <VersionSuffix>beta3</VersionSuffix>
```
`rc1`

Please also move this file to the `build` folder, so we keep everything in the same place.

PinpointTownes commented on build/common.props:
```diff
@@ -0,0 +1,20 @@
+    <VersionSuffix Condition="'$(VersionSuffix)'!='' AND '$(BuildNumber)' != ''">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)'=='netstandard1.3' AND '$(OutputType)'=='library'">
```
You can kill that, it's what causing all these warnings in the logs (because well, the `NETStandard.Library` is automatically referenced for us, so no need to have that here)

PinpointTownes commented:
Replaced by https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/pull/148.

@StummeJ thanks for your contribution.
